core/src/main/scala/kafka/server/BrokerServer.scala

```diff
@@ -627,8 +628,38 @@ class BrokerServer(
   }
 
   private def createShareCoordinator(): ShareCoordinator = {
-    val shareConfig = new ShareCoordinatorConfig(config.shareCoordinatorStateTopicSegmentBytes)
-    new ShareCoordinatorService(shareConfig)
+    val time = Time.SYSTEM
+    val serde = new RecordSerde
```

Comment:
query: Do we require different instances of Serde for Group and Share Coordinator?

Reply:
since recordserde only has non-static member methods - I think there could be synchronization issues if we use the same instance.

Reply:
Thats true, will share